### ISSUE TITLE
aws_stepfunctions_tasks: DynamoAttributeValue.fromStringSet requires an array

### ISSUE THREAD

**Issue by dbartholomae** [aws/aws-cdk#25693]
### Describe the bug

When creating a JsonPath value, it returns a string even for arrays, but DynamoAttributeValue.fromStringSet needs a string array.

```
    const releaseSlot = new DynamoUpdateItem(this, "ReleaseSlot", {
      key: {
        [partitionKey]: DynamoAttributeValue.fromString(fixedKey),
      },

      updateExpression: `DELETE ${workflowsKey} :val`,
      expressionAttributeValues: {
        ":val": DynamoAttributeValue.fromStringSet(
          JsonPath.array(
            JsonPath.stringAt("$$.Execution.Id")
          ) as unknown as string[]
        ),
      },
      table: slots,
    });
```

### Expected Behavior

This should work without the `as unknown as string[]` cast

### Current Behavior

`as unknown as string[]` cast is required

### Reproduction Steps

The code above without the cast will cause the TypeScript error.

### Possible Solution

Accept a `string` or `string[]`. This would be non-breaking.

### CDK CLI Version

2.80.0

### Node.js Version

16.14.2

### OS

Mac

### Language

Typescript

### Language Version

5.0.4

**Comment by dbartholomae** [comment]
My current implementation works on runtime and just fails on compile time. Unfortunately, your suggested solution doesn't:
https://github.com/aws/aws-cdk/blob/cdafcc52ad4aea3ef7f1446da7521fb504cb33b9/packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts#L190

JsonPath can't be used inside an array :(

You are right that unions won't work with jsii, though. Maybe an additional method?

**Comment by peterwoodworth** [comment]
You're right, I didn't actually attach the task when synthesizing before.

> Maybe an additional method?

I think it would probably be best if we can create a new method and deprecate the old one, I'm not too sure exactly what this would look like however

**Comment by dbartholomae** [comment]
How about `JsonPath.arrayString`? I'm not sure if we can deprecate the old one - if it is used at all, then the new one won't actually work in the _other_ usecase which most likely expects an array.

**Comment by dbartholomae** [comment]
This was discussed in the first CDK office hours with @mrgrain 

Should this be a p1 as type assignments won't work in other languages?
